wandb/sdk/wandb_run.py

```diff
@@ -1010,134 +1005,118 @@ def job_type(self) -> str:
         """
-        if name is None:
-            if self.settings._jupyter:
-                notebook_name = None
-                if self.settings.notebook_name:
-                    notebook_name = self.settings.notebook_name
-                elif self.settings.x_jupyter_path:
-                    if self.settings.x_jupyter_path.startswith("fileId="):
-                        notebook_name = self.settings.x_jupyter_name
-                    else:
-                        notebook_name = self.settings.x_jupyter_path
-                name_string = f"{self._settings.project}-{notebook_name}"
-            else:
-                name_string = (
-                    f"{self._settings.project}-{self._settings.program_relpath}"
-                )
-            name = wandb.util.make_artifact_name_safe(f"source-{name_string}")
-        art = wandb.Artifact(name, "code")
-        files_added = False
-        if root is not None:
-            root = os.path.abspath(root)
-            for file_path in filenames.filtered_dir(root, include_fn, exclude_fn):
-                files_added = True
-                save_name = os.path.relpath(file_path, root)
-                art.add_file(file_path, name=save_name)
-        # Add any manually staged files such as ipynb notebooks
-        for dirpath, _, files in os.walk(self._settings._tmp_code_dir):
-            for fname in files:
-                file_path = os.path.join(dirpath, fname)
-                save_name = os.path.relpath(file_path, self._settings._tmp_code_dir)
-                files_added = True
-                art.add_file(file_path, name=save_name)
-        if not files_added:
-            wandb.termwarn(
-                "No relevant files were detected in the specified directory. No code will be logged to your run."
-            )
-            return None
-
-        return self._log_artifact(art)
+        deprecate.deprecate(
+            field_name=deprecate.Deprecated.run__get_project_url,
+            warning_message=(
+                "The get_project_url method is deprecated and will be removed in a"
+                " future release. Please use `run.project_url` instead."
+            ),
+        )
+        return self.project_url
 
-    def get_project_url(self) -> str | None:
+    @property
+    @_run_decorator._attach
+    def project_url(self) -> str | None:
         """Return the url for the W&B project associated with the run, if there is one.
 
         Offline runs will not have a project url.
         """
         if self._settings._offline:
             wandb.termwarn("URL not available in offline run")
             return None
+        if self._settings._noop:
+            return None
         return self._settings.project_url
 
     def get_sweep_url(self) -> str | None:
-        """Return the url for the sweep associated with the run, if there is one."""
+        """Return the url for the sweep associated with the run, if there is one.
```

Comment:
```suggestion
        """The URL of the sweep associated with the run, if there is one.
```

+ same edit on the other docstring